.github/workflows/main.yml

```diff
@@ -0,0 +1,67 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
+      - master
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - main
+      - master
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - run: cargo check
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - run: cargo test
+
+  lints:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - run: cargo fmt --all -- --check
+      - run: cargo clippy
+
+  release:
+    needs: [check, test, lints]
+    if: ${{ startsWith(github.ref, 'refs/tags/') }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        target: [ x86_64-unknown-linux-gnu ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - run: cargo install cross
+      - run: cross build --release --target ${{ matrix.target }}
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          body: ""
+      - name: Upload Release Asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./target/${{ matrix.target }}/release/${{ github.event.repository.name }}
+          asset_name: ${{ github.event.repository.name }}-${{ matrix.target }}
+          asset_content_type: application/octet-stream
```

Comment:
Example Release:

https://github.com/gausnes/prometheus_wireguard_exporter/releases/tag/v0.0.2

Was able to download and start the agent on the pipe-dev node:

```shell
$ wget https://github.com/gausnes/prometheus_wireguard_exporter/releases/download/v0.0.2/prometheus_wireguard_exporter-x86_64-unknown-linux-gnu
$ chmod +x prometheus_wireguard_exporter-x86_64-unknown-linux-gnu
$ ./prometheus_wireguard_exporter-x86_64-unknown-linux-gnu
[2021-11-01T23:50:37Z INFO  prometheus_wireguard_exporter] prometheus_wireguard_exporter v3.5.1 starting...
[2021-11-01T23:50:37Z INFO  prometheus_wireguard_exporter] using options: Options { verbose: false, prepend_sudo: false, extract_names_config_files: None, interfaces: None, metric_attributes: MetricAttributeOptions { split_allowed_ips: false, export_remote_ip_and_port: false, handshake_timeout_seconds: None } }
[2021-11-01T23:50:37Z INFO  prometheus_wireguard_exporter] starting exporter on http://0.0.0.0:9586/metrics
[2021-11-01T23:50:37Z INFO  prometheus_exporter_base] Listening on http://0.0.0.0:9586/metrics
```